source/simulator/assemblers/stokes.cc

```diff
@@ -229,11 +229,15 @@ namespace aspect
               data.local_rhs(i) += (density * gravity * scratch.phi_u[i])
                                    * JxW;
 
-              if (force != NULL)
+              if (force != NULL && this->get_parameters().enable_additional_stokes_rhs)
                 data.local_rhs(i) += (force->rhs_u[q] * scratch.phi_u[i]
                                       + pressure_scaling * force->rhs_p[q] * scratch.phi_p[i])
                                      * JxW;
 
+              if (force != NULL && this->get_parameters().enable_elasticity)
```

Comment:
If you move `rhs_e` out of `AdditionalMaterialOutputsStokesRHS` and into `ElasticOutputs` then you can create a pointer `elastic_outputs` and replace this condition by `if (elastic_outputs != NULL)`